python/ray/rllib/rollout.py

```diff
@@ -102,17 +104,31 @@ def run(args, parser):
     rollout(agent, args.env, num_steps, args.out, args.no_render)
 
 
+class DefaultMapping(collections.defaultdict):
+    """default_factory now takes as an argument the missing key."""
+
+    def __missing__(self, key):
+        self[key] = value = self.default_factory(key)
+        return value
+
+
 def rollout(agent, env_name, num_steps, out=None, no_render=True):
     if hasattr(agent, "local_evaluator"):
         env = agent.local_evaluator.env
-        multiagent = agent.local_evaluator.multiagent
-        if multiagent:
+        multiagent = isinstance(env, MultiAgentEnv)
+        if agent.local_evaluator.multiagent:
             policy_agent_mapping = agent.config["multiagent"][
                 "policy_mapping_fn"]
-            mapping_cache = {}
+        else:
+            policy_agent_mapping = lambda _: DEFAULT_POLICY_ID
+        mapping_cache = {}  # in case policy_agent_mapping is stochastic
         policy_map = agent.local_evaluator.policy_map
         state_init = {p: m.get_initial_state() for p, m in policy_map.items()}
         use_lstm = {p: len(s) > 0 for p, s in state_init.items()}
+        action_init = {
+            p: m.action_space.sample()
```

Comment:
note that in the actual code we feed the all-zeros action initially

Reply:
Does that make sense? All zeros might not even be in the action space?

Reply:
Yeah that's a good question.

Reply:
I think this is fine for now, though it would be better to set it to zeros for consistency (or switch the sampler to stick in a random initial action, but that might be weird).